src/utils/validation/validateSync/applyFormRules.js

```diff
@@ -21,6 +22,9 @@ function reduceRules(rules) {
 export default function applyFormRules(rules) {
   return (resolverArgs) => {
     const rulesList = listOf(
+      /** @todo This is weird */
+      addWhen(R.path(['fieldPath', 'name'], rules), isset, reduceRules),
+      addWhen(R.path(['fieldPath', 'type'], rules), isset, reduceRules),
```

Comment:
There is no exclusion between grouped rules and root-level rules.

I think that group-specific rules (both name and type) must always overwrite the root-level rules. So that the behavior is not accumulative between grouped rules and root rules.